src/Data/Array/Mutable/Linear.hs

```diff
@@ -215,6 +220,20 @@ slice from targetSize arr =
     wrap :: (# Array# a, Array# a  #) #-> (Array a, Array a)
     wrap (# old, new #) = (Array old, Array new)
 
+-- | /O(1)/ Convert an 'Array' to an immutable 'Vector.Vector' (from
+-- 'vector' package).
+freeze :: Array a #-> Ur (Vector.Vector a)
+freeze (Array arr) =
+  Unlifted.freeze go arr
+ where
+   go arr = unsafeDupablePerformIO $ do
+     mut <- Prim.unsafeThawArray (Prim.Array arr)
+     let mv = MVector.MVector 0 (Prim.sizeofMutableArray mut) mut
+     Vector.unsafeFreeze mv
+   -- We only need to do above because 'Vector' constructor is hidden.
+   -- Once it is exposed, we should be able to replace it with something
+   -- safer like: `go arr = Vector 0 (sizeof arr) arr`
```

Comment:
Let's make an issue in our tracker to fix this. And we'll contribute upstream some way. After milestone 1.

Reply:
I opened: https://github.com/tweag/linear-base/issues/214